contracts/deposit-service/DepositServiceBase.sol

```diff
@@ -14,21 +14,23 @@ contract DepositServiceBase is IDepositServiceBase {
 
     // Using immutable storage to keep the constants in the bytecode
     address public immutable gateway;
+    address public immutable wrappedTokenAddress;
     bytes32 internal immutable wrappedSymbolBytes;
 
     constructor(address gateway_, string memory wrappedSymbol_) {
         if (gateway_ == address(0)) revert InvalidAddress();
 
         gateway = gateway_;
+        wrappedTokenAddress = IAxelarGateway(gateway_).tokenAddresses(wrappedSymbol_);
 
         // Checking if token symbol exists in the gateway
-        if (IAxelarGateway(gateway_).tokenAddresses(wrappedSymbol_) == address(0)) revert InvalidSymbol();
+        if (bytes(wrappedSymbol_).length > 0 && wrappedTokenAddress == address(0)) revert InvalidSymbol();
```

Comment:
```suggestion
        if (bytes(wrappedSymbol_).length > 0) {
            wrappedTokenAddress = IAxelarGateway(gateway_).tokenAddresses(wrappedSymbol_);
            if (wrappedTokenAddress == address(0)) revert InvalidSymbol();
        }
```

maybe something like this instead?

Reply:
This pattern throws a compiler error. But I will add the comment to make that line more clear
```
TypeError: Cannot write to immutable here: 
Immutable variables cannot be initialized inside an if statement.
```